tests/containers/bci_repo.pm

```diff
@@ -42,41 +42,55 @@ my $tdata = [
     }
 ];
 
+my @pkg_regex = ('kernel-default', 'yast', 'gnome-desktop', 'qemu-kvm');
+
 sub container_exec {
     my $container = shift;
     my $cmd = shift;
     assert_script_run(qq[podman exec $container /bin/sh -c '$cmd'], @_);
 }
 
+sub prepare_repo {
+    my $container = shift;
+    my $bci_repo = get_required_var('REPO_BCI');
+    container_exec($container, 'zypper ref', timeout => 180);
+    container_exec($container, 'zypper lr -d');
+    container_exec($container, "zypper -q -s 11 pa --orphaned | tee -a repo.org", timeout => 600);
+    # remove container-suseconnect to not get packages from registered SLE host
+    container_exec($container, 'zypper -n rm container-suseconnect', timeout => 180);
+    # refresh services in order to remove container-suseconnect-zypp orphaned services
+    container_exec($container, 'zypper refs');
+    # remove SLE_BCI repo pointing to official update servers and add SUT repo
+    container_exec($container, 'zypper rr 1');
+    container_exec($container, "zypper ar http://openqa.suse.de/assets/repo/$bci_repo BCI_TEST");
+    container_exec($container, 'zypper ref', timeout => 180);
+    container_exec($container, 'zypper lr -d');
+}
+
 sub run {
     my ($self) = @_;
     $self->select_serial_terminal;
-    my $bci_repo = get_required_var('REPO_BCI');
 
     # download and start a BCI container
     my $image = get_required_var('CONTAINER_IMAGE_TO_TEST');
     record_info('IMAGE', $image);
     script_retry("podman pull $image", timeout => 300, delay => 60, retry => 3);
     record_info('Inspect', script_output("podman inspect $image"));
 
+    record_info('TEST', 'Test that the repo does not contain certain packages such as kernel, yast, desktop, kvm, etc...');
+    my $container = "bci-repo-tester_pkgs";
+    assert_script_run("podman run --name $container -dt $image");
+    prepare_repo($container);
+    foreach (@pkg_regex) {
+        my $pkg = $_;
+        die ("Package $pkg should not be present in BCI-repo!") if (script_run(qq[podman exec $container /bin/sh -c 'zypper se $pkg'], die_on_timeout => 1) == 0);
```

Comment:
This means it dies on the first found issue already. IMO it's better to continue looking and at the end, show a list of all issues at once.